Adding tox.ini and Django3.2 support

saksham115 opened the pull request:
This PR introduces two functionalities:

1. Ability to run tests using `tox`
2. Running tests on django-3.2

mtyaka commented on README.md:
```diff
@@ -7,3 +7,7 @@
+
+## Testing
+
+1. For running the tests on Python 3.6 and Django versions 2.2 and 3.2 use the command `tox`
```
Nit: I don't think you need to specify the versions here, "For running the tests use the command `tox`" should be clear enough.

mtyaka commented on tox.ini:
```diff
@@ -0,0 +1,16 @@
+[tox]
+envlist = py36-django{22,32}
```
Can we also add `py38-django{22,32} to this list? edX has already migrated all their services to Python 3.8, so we should make sure this xblock works on 3.8 as well.

mtyaka reviewed (commented):
@saksham115 Thanks for this! The changes look good, but it would be great if we could also test on python 3.8. Hopefully with tox that should be relatively simple.

Unfortunately it looks like Travis CI integration is not working anymore. I'll try to get it restored, or perhaps switch to Circle CI.

For future PRs, please include step-by-step testing instructions as per [the PR template](https://github.com/edx/edx-platform/blob/master/.github/pull_request_template.md) -- it can save a lot of reviewer time on more complex PRs (it's not really required on this PR which is simple enough, but it's a good habit to get into :) )

mtyaka commented:
Thanks @saksham115, the updates look good, but when looking at the Circle CI log of the [latest build](https://app.circleci.com/pipelines/github/open-craft/xblock-vectordraw/20/workflows/26ac24f0-b3eb-4f31-99b9-4be39e251647/jobs/50) I can see that the tests actually run with django 2.2. Tox does seem to first instal 3.2, but 3.2 later gets uninstalled and replaced with 2.2, apparently because `xblock-sdk/requirements/base.txt` pins Django to 2.2.

I'm trying to understand how xblock-sdk manage to run their tests with different Django versions, and it looks like they don't install the base.txt requirements, just the test.text one (see [here](https://github.com/edx/xblock-sdk/blob/3d96586dd029ff3cb44a832499c83e0a0ec5a9d1/tox.ini)).

Can you try removing the commands that install `xblock-sdk/requirements/base.txt` requirements and see if that fixes it? I am guessing we don't need to explicitly install requirements form `xblock/requirements/*` either -- can you try removing that from the `tox.ini` file as well?

saksham115 commented:
> Thanks @saksham115, the updates look good, but when looking at the Circle CI log of the [latest build](https://app.circleci.com/pipelines/github/open-craft/xblock-vectordraw/20/workflows/26ac24f0-b3eb-4f31-99b9-4be39e251647/jobs/50) I can see that the tests actually run with django 2.2. Tox does seem to first instal 3.2, but 3.2 later gets uninstalled and replaced with 2.2, apparently because `xblock-sdk/requirements/base.txt` pins Django to 2.2.
> 
> I'm trying to understand how xblock-sdk manage to run their tests with different Django versions, and it looks like they don't install the base.txt requirements, just the test.text one (see [here](https://github.com/edx/xblock-sdk/blob/3d96586dd029ff3cb44a832499c83e0a0ec5a9d1/tox.ini)).
> 
> Can you try removing the commands that install `xblock-sdk/requirements/base.txt` requirements and see if that fixes it? I am guessing we don't need to explicitly install requirements form `xblock/requirements/*` either -- can you try removing that from the `tox.ini` file as well?

Definitely, will give it a shot

saksham115 commented:
> Thanks @saksham115, the updates look good, but when looking at the Circle CI log of the [latest build](https://app.circleci.com/pipelines/github/open-craft/xblock-vectordraw/20/workflows/26ac24f0-b3eb-4f31-99b9-4be39e251647/jobs/50) I can see that the tests actually run with django 2.2. Tox does seem to first instal 3.2, but 3.2 later gets uninstalled and replaced with 2.2, apparently because `xblock-sdk/requirements/base.txt` pins Django to 2.2.
> 
> I'm trying to understand how xblock-sdk manage to run their tests with different Django versions, and it looks like they don't install the base.txt requirements, just the test.text one (see [here](https://github.com/edx/xblock-sdk/blob/3d96586dd029ff3cb44a832499c83e0a0ec5a9d1/tox.ini)).
> 
> Can you try removing the commands that install `xblock-sdk/requirements/base.txt` requirements and see if that fixes it? I am guessing we don't need to explicitly install requirements form `xblock/requirements/*` either -- can you try removing that from the `tox.ini` file as well?

@mtyaka you were right.

I went into the virtual environment for `py38-django32` that was setup by tox and did `pip freeze`. Django 2.2 was getting installed.

As per your suggestion, I removed the `base.txt` install and then django 3.2 was installing. 🙌🏻 
I have made the fix and pushed the changes

mtyaka commented:
Thanks for your patience @saksham115! This is good to merge 👍 

After you merge, please create and push a new v0.4.0 [tag](https://github.com/open-craft/xblock-vectordraw/tags).

- [x] I tested this: I verified that he tests pass on Django 3.2
- [x] I read through the code
- [ ] ~~I checked for accessibility issues~~
- [x] Includes documentation
